### PR TITLE
RadioButton listener added

### PR DESCRIPTION
**Pull request by Vishal1707**
type
  RadioButtons* = ref object of Widget
    impl*: ptr rawui.RadioButtons
    onradiobuttonclick*: proc() {.closure.}

voidCallback(wrapOnRadioButtonClick,RadioButtons,RadioButtons,onradiobuttonclick)

proc add*(r: RadioButtons; text: string) =
  radioButtonsAppend(r.impl, text)

proc radioButtonsSelected*(r: RadioButtons): int =
  radioButtonsSelected(r.impl)

### CODE REVIEW

File: ui.nim [hunk 1]
```diff
@@ -29,23 +29,22 @@ proc pollingMainLoop*(poll: proc(timeout: int); timeout: int) =
...
   rawui.uninit()
 
-# ------------------- Button --------------------------------------
-
 template newFinal(result) =
   #proc finalize(x: type(result)) {.nimcall.} =
   #  controlDestroy(x.impl)
   new(result) #, finalize)
-
+
```
Comment (Araq) [inline ui.nim]: whitespace added, remove it.

File: ui.nim [hunk 2]
```diff
@@ -29,23 +29,22 @@ proc pollingMainLoop*(poll: proc(timeout: int); timeout: int) =
...
   #  controlDestroy(x.impl)
   new(result) #, finalize)
-
+  
 template voidCallback(name, supertyp, basetyp, on) {.dirty.} =
   proc name(w: ptr rawui.supertyp; data: pointer) {.cdecl.} =
     let widget = cast[basetyp](data)
     if widget.on != nil: widget.on()
-
+
```
Comment (Araq) [inline ui.nim]: same

**Comment by dom96** [comment]
Needed a contribution for today so decided to fix this for @Vishal1707  :)

**Comment by Vishal1707** [comment]
Thanks @dom96.

**Pull request merged**